lib/child-writer/index.js

```diff
@@ -165,59 +172,222 @@ exports.write = (image, drive, options) => {
+    ipc.server.start()
 
-    child.on('exit', (code, signal) => {
-      terminateServer()
+    return this
+  }
 
-      if (code === EXIT_CODES.CANCELLED) {
-        return emitter.emit('done', {
-          cancelled: true
-        })
-      }
+  /**
+   * @summary Detach event handlers from the IPC
+   * and child to avoid emitting further events
+   * @private
+   *
+   * @example
+   * this.detachHandlers()
+   */
+  detachHandlers () {
+    // NOTE: The `ipc.server` is not an instance of EventEmitter,
+    // but a PubSub; see https://github.com/RIAEvangelist/event-pubsub
+    // Thus the method to remove event handlers is not `.removeListener()`
+    ipc.server.off('error', this.onError)
+    ipc.server.off('message', this.bridgeRobotMessage)
+    this.child.removeListener('exit', this.onExit)
+  }
+
+  /**
+   * @summary Abort the current writing process
+   * @public
+   *
+   * @param {Function} callback - callback
+   *
+   * @example
+   * childWriter.abort(() => {
+   *   console.log('Writing aborted')
+   * })
+   */
+  abort (callback) {
+    debug('abort')
 
-      // We shouldn't emit the `done` event manually here
-      // since the writer process will take care of it.
-      if (code === EXIT_CODES.SUCCESS || code === EXIT_CODES.VALIDATION_ERROR) {
-        return null
+    if (this.aborting) {
+      debug('abort:in-progress')
+      if (_.isFunction(callback)) {
+        this.once('close', callback)
       }
+      return
+    }
 
-      const error = new Error(`Child process exited with code ${code}, signal ${signal}`)
-      error.code = code
-      error.signal = signal
+    this.aborting = true
+    this.detachHandlers()
+    this.child.once('exit', (code, signal) => {
+      debug('abort:child:exit', code, signal)
+      this.emit('done', {
+        cancelled: true
+      })
+      this.terminateServer(callback)
+    })
 
-      return emitError(error)
+    ipc.server.broadcast('abort', {
+      signal: 'SIGINT'
     })
-  })
+  }
+}
 
-  ipc.server.start()
+/**
+ * @summary The currently running instance of the writer
+ * @type {ChildWriter}
+ */
+ChildWriter.currentInstance = null
+
+/**
+ * @summary Perform a write
+ * @function
+ * @public
+ *
+ * @param {String} image - image
+ * @param {Object} drive - drive
+ * @param {Object} options - options
+ * @returns {ChildWriter} childWriter
+ *
+ * @example
+ * const child = childWriter.write('path/to/rpi.img', {
+ *   device: '/dev/disk2'
+ * }, {
+ *   validateWriteOnSuccess: true,
+ *   unmountOnSuccess: true
+ * })
+ *
+ * child.on('progress', (state) => {
+ *   console.log(state)
+ * })
+ *
+ * child.on('error', (error) => {
+ *   throw error
+ * })
+ *
+ * child.on('done', () => {
+ *   console.log('Validation was successful!')
+ * })
+ */
+ChildWriter.write = (image, drive, options) => {
+  if (ChildWriter.currentInstance) {
+    throw new Error('Write already in progress')
+  }
+  const writer = new ChildWriter().write(image, drive, options)
```

Comment:
Not something we have to deal with right before the release, but why the extra complexity of making this a class, when it could be modeled as a set of mostly stateless functions? I see your subclassing EventEmitter, but looks like we can use composition instead of inheritance as well.